c/contracts/meta_contract.c

```diff
@@ -97,6 +99,64 @@ int main() {
       ckb_debug("failed to set return data");
       return ret;
     }
+  } else if (msg.item_id == MSG_BATCH_CREATE_ACCOUNTS) {
+    /* charge fee */
+    mol_seg_t fee_seg = MolReader_BatchCreateEthAccounts_get_fee(&msg.seg);
+    mol_seg_t amount_seg = MolReader_Fee_get_amount(&fee_seg);
+    mol_seg_t reg_id_seg = MolReader_Fee_get_registry_id(&fee_seg);
+
+    uint256_t fee_amount = {0};
+    _gw_fast_memcpy((uint8_t *)(&fee_amount), (uint8_t *)amount_seg.ptr,
+                    sizeof(uint128_t));
+
+    uint32_t reg_id = *(uint32_t *)reg_id_seg.ptr;
```

Comment:
Potentially unaligned pointer cast and load? Use something like https://github.com/nervosnetwork/molecule/blob/5f94aa86d523d60e498ffefe01001fcf1ae70851/bindings/c/include/molecule_reader.h#L84-L99 instead?

Reply:
Another approach is `memcpy`, but we already use too many cast in the C code.

Reply:
Good catch. I think we could add `-fsanitize=undefined`, we need a new toolchain image.